metrics.py

```diff
@@ -108,14 +112,18 @@ def calc_change_score(current_network: np.ndarray, indices: tuple):
         return -1 if current_network[indices[0], indices[1]] else 1
 
     @staticmethod
-    @njit
-    def calculate_for_sample(networks_sample: np.ndarray):
+    # @njit
```

Comment:
Doesn't numba work with torch? If so, we may want to implement a custom decorator that decides to use njit only when the input is a numpy array (I think it is possible)

Reply:
it throws an error, but yeah we can do it as a custom decorator

Reply:
So the custom decorator solution is kind of ugly, and it seems that a better thing to do is split into two functions (also suggested by the chat)

<img width="959" alt="image" src="https://github.com/user-attachments/assets/beb73801-f923-4d65-b8fc-f27889829085">

Reply:
Fine